### PR TITLE
Enable static build option for color model functions

### PR DESCRIPTION
**Pull request by r-abishek**

### CODE REVIEW

**Comment by kiritigowda** [comment]
@r-abishek conflicts need to be resolved

File: src/modules/hip/kernel/hue.cpp [hunk 1]
```diff
@@ -123,200 +174,325 @@ extern "C" __global__ void huergb_pkd(     unsigned char *input,
...
+        hsv = convert_one_pixel_to_hsv(pixel);
+        hsv.y *= sat[id_z];
 
-  if ((id_y >= yroi_begin[id_z]) && (id_y <= yroi_end[id_z]) &&
-    (id_x >= xroi_begin[id_z]) && (id_x <= xroi_end[id_z])) {
-    hsv = convert_one_pixel_to_hsv(pixel);
-    hsv.y *= sat[id_z];
-    if (hsv.y > 1.0) {
-      hsv.y = 1.0;
+        if (hsv.y > 1.0)
```
Comment (rrawther) [inline src/modules/hip/kernel/hue.cpp:324]: better to use min and max instruction here to avoid conditional branch

**Review by rrawther — CHANGES_REQUESTED** [review]
@pavel please make sure the tests pass before merging

File: src/modules/hip/kernel/hue.cpp (hunk 1 above)
Comment (r-abishek) [reply src/modules/hip/kernel/hue.cpp]: Modified to use HIP's fmaxf, fminf

**Comment by r-abishek** [comment]
> @r-abishek conflicts need to be resolved

@kiritigowda Should be resolved now

**Review by asalmanp — APPROVED** [review]

**Comment by pavel** [comment]
@rrawther Thanks for notifying me, but it does not seem that I'm involved in this project in any way.

https://blog.github.com/2011-03-23-mention-somebody-they-re-notified/

**Pull request merged**